src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -28,46 +38,207 @@ const ActorInit Obj_Lift_InitVars = {
     (ActorFunc)ObjLift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8093DD84[] = {
-    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_CONTINUE),
-    ICHAIN_F32_DIV1000(terminalVelocity, -15000, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneScale, 350, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneDownward, 350, ICHAIN_STOP),
-};
-
-#endif
-
-extern InitChainEntry D_8093DD84[];
-
-extern UNK_TYPE D_06000D10;
-extern UNK_TYPE D_06000F00;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D3C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/ObjLift_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/ObjLift_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D760.s")
+static s16 D_8093DD50[] = { 0, 10, 20, 30, 40, 50, 60 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D7A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D88C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D8B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D9C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DA48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DB90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/ObjLift_Update.s")
+Vec2s D_8093DD60[] = {
+    { 120, -120 }, { 120, 0 },     { 120, 120 }, { 0, -120 },   { 0, 0 },
+    { 0, 120 },    { -120, -120 }, { -120, 0 },  { -120, 120 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/ObjLift_Draw.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_CONTINUE),   ICHAIN_F32_DIV1000(terminalVelocity, -15000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneScale, 350, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 350, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DC90.s")
+static f32 D_8093DD98[] = { 0.1f, 0.05f };
+
+static f32 yOffset[] = { -18.0f, -9.0f };
+
+void func_8093D3C0(ObjLift* this, PlayState* play) {
+    Vec3f pos;
+    Vec3f vel;
+    Vec3f* actorPos = &this->dyna.actor.world.pos;
+    s32 i;
+    s32 rnd;
```

Comment:
```suggestion
    s32 rand;
```
`rand` is much more common, we try not to drop vowels in names if possible to keep it readable